components/state/src/index.js

```diff
@@ -96,7 +98,7 @@ AFRAME.registerSystem('state', {
       }
 
       // Notify subscriptions / binders.
-      toUpdate.length = 0;
```

Comment:
maybe this style of array clearing is still okay to use? might be even faster to have less array modification

Reply:
When we call ```subscriber.onStateUpdate``` and it updates the state inside this method, we will again get inside this method and modify ```toUpdate``` array. Thus we don't want to completely clean up the array, but remove only the subscriber we're notifying right now. Then we need to do something like ```toUpdate.length=toUpdate.length-1```. But ```pop``` does the exact same thing under the hood, as far as I'm concerned.